### ISSUE TITLE
[BUG] java.lang.AssertionError: assertion failed: inconsistent record domains of size 1 and 2

### ISSUE THREAD

**Issue by vitorenesduarte** [informalsystems/apalache#627]
**To Reproduce**
```
git clone https://github.com/vitorenesduarte/Examples -b apalache_2pc
cd Examples/specifications/transaction_commit/
apalache-mc check TwoPhase.tla
```

```
# APALACHE version 0.11.0 build v0.10.0-30-g8c04870
#
# WARNING: This tool is in the experimental stage.
#          Please report bugs at: [https://github.com/informalsystems/apalache/issues]
#
# Usage statistics is OFF. We care about your privacy.
# If you want to help our project, consider enabling statistics with config --enable-stats=true.

Checker options: filename=TwoPhase.tla, init=, next=, inv=        I@11:53:52.222
Tuning:                                                           I@11:53:52.411
PASS #0: SanyParser                                               I@11:53:52.412
Parsing file /home/vitor.enes/Examples/specifications/transaction_commit/TwoPhase.tla
Parsing file /home/vitor.enes/Examples/specifications/transaction_commit/TCommit.tla
PASS #1: ConfigurationPass                                        I@11:53:52.715
  > TwoPhase.cfg: Loading TLC configuration                       I@11:53:52.717
  > Using the init predicate TPInit from the TLC config           I@11:53:52.770
  > Using the next predicate TPNext from the TLC config           I@11:53:52.770
  > TwoPhase.cfg: found INVARIANTS: TPTypeOK                      I@11:53:52.770
  > Set the initialization predicate to TPInit                    I@11:53:52.771
  > Set the transition predicate to TPNext                        I@11:53:52.771
  > Set an invariant to TPTypeOK                                  I@11:53:52.772
  > Replaced CONSTANT RM with {"r1", "r2", "r3"}                  I@11:53:52.773
PASS #2: DesugarerPass                                            I@11:53:52.797
  > Desugaring...                                                 I@11:53:52.797
PASS #3: UnrollPass                                               I@11:53:52.816
  > Unroller                                                      I@11:53:52.839
PASS #4: PrimingPass                                              I@11:53:52.853
  > Introducing TPInitPrimed for TPInit'                          I@11:53:52.855
PASS #5: CoverAnalysisPass                                        I@11:53:52.866
PASS #6: InlinePass                                               I@11:53:52.867
  > InlinerOfUserOper                                             I@11:53:52.871
  > LetInExpander                                                 I@11:53:52.882
  > InlinerOfUserOper                                             I@11:53:52.886
Leaving only relevant operators: CInitPrimed, TPInit, TPInitPrimed, TPNext, TPTypeOK I@11:53:52.888
PASS #7: VCGen                                                    I@11:53:52.896
  > Producing verification conditions from the invariant TPTypeOK I@11:53:52.896
  > VCGen produced 4 verification condition(s)                    I@11:53:52.897
PASS #8: PreprocessingPass                                        I@11:53:52.908
  > Before preprocessing: unique renaming                         I@11:53:52.908
 > Applying standard transformations:                             I@11:53:52.911
  > PrimePropagation                                              I@11:53:52.912
  > Desugarer                                                     I@11:53:52.924
  > UniqueRenamer                                                 I@11:53:52.936
  > Normalizer                                                    I@11:53:52.954
  > Keramelizer                                                   I@11:53:52.966
  > After preprocessing: UniqueRenamer                            I@11:53:52.986
PASS #9: TransitionFinderPass                                     I@11:53:53.003
  > Found 1 initializing transitions                              I@11:53:53.023
  > Found 7 transitions                                           I@11:53:53.039
  > No constant initializer                                       I@11:53:53.040
  > Applying unique renaming                                      I@11:53:53.040
PASS #10: OptimizationPass                                        I@11:53:53.053
 > Applying optimizations:                                        I@11:53:53.055
  > ConstSimplifier                                               I@11:53:53.056
  > ExprOptimizer                                                 I@11:53:53.059
  > ConstSimplifier                                               I@11:53:53.062
PASS #11: AnalysisPass                                            I@11:53:53.078
 > Marking skolemizable existentials and sets to be expanded...   I@11:53:53.079
  > SkolemizationMarker                                           I@11:53:53.079
  > ExpansionMarker                                               I@11:53:53.080
 > Running analyzers...                                           I@11:53:53.084
  > Introduced expression grades                                  I@11:53:53.091
  > Introduced 7 formula hints                                    I@11:53:53.092
PASS #12: BoundedChecker                                          I@11:53:53.092
State 0: Checking 4 invariants                                    I@11:53:53.306
Step 0: picking a transition out of 1 transition(s)               I@11:53:53.360
Step 1: Transition #0 is disabled                                 I@11:53:53.367
State 1: Checking 2 invariants                                    I@11:53:53.372
Step 1: Transition #2 is disabled                                 I@11:53:53.408
Step 1: Transition #3 is disabled                                 I@11:53:53.416
Step 1: Transition #4 is disabled                                 I@11:53:53.422
State 1: Checking 2 invariants                                    I@11:53:53.433
State 1: Checking 1 invariants                                    I@11:53:53.457
Step 1: picking a transition out of 3 transition(s)               I@11:53:53.461
(Please report an issue at: [https://github.com/informalsystems/apalache/issues],java.lang.AssertionError: assertion failed: inconsistent record domains of size 1 and 2)
Unhandled exception                                               E@11:53:53.469
java.lang.AssertionError: assertion failed: inconsistent record domains of size 1 and 2
	at scala.Predef$.assert(Predef.scala:223)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.$anonfun$pickRecordDomain$1(CherryPick.scala:278)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.$anonfun$pickRecordDomain$1$adapted(CherryPick.scala:275)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickRecordDomain(CherryPick.scala:275)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickRecord(CherryPick.scala:237)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickByOracle(CherryPick.scala:121)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickOneElement$1(CherryPick.scala:371)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.$anonfun$pickSetNonEmpty$12(CherryPick.scala:394)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickSetNonEmpty(CherryPick.scala:394)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickSet(CherryPick.scala:324)
	at at.forsyte.apalache.tla.bmcmt.rules.aux.CherryPick.pickByOracle(CherryPick.scala:124)
	at at.forsyte.apalache.tla.bmcmt.trex.TransitionExecutorImpl.pickVar$1(TransitionExecutorImpl.scala:240)
	at at.forsyte.apalache.tla.bmcmt.trex.TransitionExecutorImpl.$anonfun$pickTransition$4(TransitionExecutorImpl.scala:247)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:285)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:285)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:278)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at at.forsyte.apalache.tla.bmcmt.trex.TransitionExecutorImpl.pickTransition(TransitionExecutorImpl.scala:247)
	at at.forsyte.apalache.tla.bmcmt.trex.FilteredTransitionExecutor.pickTransition(FilteredTransitionExecutor.scala:104)
	at at.forsyte.apalache.tla.bmcmt.SeqModelChecker.prepareTransitionsAndCheckInvariants(SeqModelChecker.scala:171)
	at at.forsyte.apalache.tla.bmcmt.SeqModelChecker.makeStep(SeqModelChecker.scala:51)
	at at.forsyte.apalache.tla.bmcmt.SeqModelChecker.run(SeqModelChecker.scala:39)
	at at.forsyte.apalache.tla.bmcmt.passes.BoundedCheckerPassImpl.runIncrementalChecker(BoundedCheckerPassImpl.scala:108)
	at at.forsyte.apalache.tla.bmcmt.passes.BoundedCheckerPassImpl.execute(BoundedCheckerPassImpl.scala:83)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.exec$1(PassChainExecutor.scala:22)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.run(PassChainExecutor.scala:37)
	at at.forsyte.apalache.tla.Tool$.runCheck(Tool.scala:173)
	at at.forsyte.apalache.tla.Tool$.$anonfun$run$2(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.$anonfun$run$2$adapted(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.handleExceptions(Tool.scala:258)
	at at.forsyte.apalache.tla.Tool$.run(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.main(Tool.scala:45)
	at at.forsyte.apalache.tla.Tool.main(Tool.scala)
It took me 0 days  0 hours  0 min  1 sec                          I@11:53:53.470
Total time: 1.249 sec                                             I@11:53:53.470
EXITCODE: ERROR (99)
```

**Desktop:**
 - Debian GNU/Linux 10
- openjdk 11.0.9.1 2020-11-04
- APALACHE version 0.11.0 build v0.10.0-30-g8c04870

**Additional context**
I'm trying to use Apalache to check the following spec of 2PC: https://github.com/tlaplus/Examples/blob/master/specifications/transaction_commit/TwoPhase.tla

I've added some annotations until I hit the issue above: https://github.com/vitorenesduarte/Examples/commit/ae38069fd9d673684c025f930abcdd7ad82c7740
I'm not sure if all are needed, or if they're even correct; sometimes it's not clear from which TLA line the error comes from (for example, running Apalache on the original file without annotations gives no hints on what to do).

**Comment by konnov** [comment]
Did you introduce any type annotations? It should not have worked without annotations. That is what happened

**Comment by konnov** [comment]
NVM. I can reproduce your report from the repo.

**Comment by konnov** [comment]
@vitorenesduarte this bug report concerns the old type annotations. Do you like to try it with 0.15.0 and see, whether the bug is still there?

**Comment by konnov** [comment]
Related to #401

**Comment by konnov** [comment]
Closing this one, as we are planning to fix records in #401.
